src/overlays/actors/ovl_Door_Spiral/z_door_spiral.c

```diff
@@ -31,30 +40,223 @@ const ActorInit Door_Spiral_InitVars = {
+    { object_danpei_object_DL_002110, object_danpei_object_DL_0012C0, 0, 130, 12, 50, 15 },
+};
+
+typedef struct {
+    /* 0x0 */ s16 objectId;
+    /* 0x2 */ u8 index;
+} SpiralObjectInfo; // size = 0x4
+
+SpiralObjectInfo sSpiralObjectInfoTable[] = {
+    { GAMEPLAY_KEEP, 0 },    { GAMEPLAY_DANGEON_KEEP, 1 }, { OBJECT_NUMA_OBJ, 2 },      { OBJECT_HAKUGIN_OBJ, 4 },
+    { OBJECT_IKANA_OBJ, 5 }, { OBJECT_DANPEI_OBJECT, 7 },  { OBJECT_IKNINSIDE_OBJ, 6 },
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/func_809A2B60.s")
+typedef struct {
+    /* 0x0 */ s16 sceneId;
+    /* 0x2 */ u8 index;
+} SpiralSceneInfo; // size = 0x4
+
+SpiralSceneInfo sSpiralSceneInfoTable[] = {
+    { SCENE_MITURIN, 2 },     { SCENE_HAKUGIN, 3 },   { SCENE_INISIE_N, 4 }, { SCENE_INISIE_R, 4 },
+    { SCENE_DANPEI2TEST, 5 }, { SCENE_IKNINSIDE, 6 }, { SCENE_CASTLE, 6 },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP),
+};
+
+void DoorSpiral_SetupAction(DoorSpiral* this, DoorSpiralActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+    this->unk14A = 0;
+}
+
+s32 func_809A2B70(DoorSpiral* this, PlayState* play) {
+    SpiralObjectInfo* spiralObjectInfo = &sSpiralObjectInfoTable[this->unk147];
+
+    this->unk148 = spiralObjectInfo->index;
+    if ((this->unk148 == 7) || ((this->unk148 == 2) && play->roomCtx.curRoom.enablePosLights)) {
+        if (this->unk148 == 2) {
+            this->unk148 = 3;
+        }
+        this->actor.flags |= ACTOR_FLAG_10000000;
+    }
+    DoorSpiral_SetupAction(this, func_809A2FF8);
+    return 0;
+}
+
+u8 func_809A2BF8(PlayState* play) {
+    SpiralSceneInfo* spiralSceneInfo = sSpiralSceneInfoTable;
+    s32 i;
+    u8 var_v1;
+
+    for (i = 0; i < ARRAY_COUNT(sSpiralSceneInfoTable); i++) {
+        if (play->sceneId == spiralSceneInfo->sceneId) {
+            break;
+        }
+        spiralSceneInfo++;
+    }
+    if (i < ARRAY_COUNT(sSpiralSceneInfoTable)) {
+        var_v1 = spiralSceneInfo->index;
+    } else {
+        var_v1 = (Object_GetSlot(&play->objectCtx, GAMEPLAY_DANGEON_KEEP) >= 0) ? 1 : 0;
+    }
+    return var_v1;
+}
+
+void DoorSpiral_Init(Actor* thisx, PlayState* play) {
+    DoorSpiral* this = THIS;
+    s32 transitionActorId = DOORSPIRAL_GET_FC00(thisx);
+    s8 temp_v0_2;
+
+    if (this->actor.room != play->doorCtx.transitionActorList[transitionActorId].sides[0].room) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->unk145 = DOORSPIRAL_GET_TYPE(thisx);
+    this->unk146 = DOORSPIRAL_GET_80(thisx);
+    this->unk147 = func_809A2BF8(play);
+    temp_v0_2 = Object_GetSlot(&play->objectCtx, sSpiralObjectInfoTable[this->unk147].objectId);
+    this->objSlot = temp_v0_2;
```

Comment:
```suggestion
    objectSlot = Object_GetSlot(&play->objectCtx, sSpiralObjectInfoTable[this->unk147].objectId);
    this->objectSlot = objectSlot;
```
If this is needed, then the temp should be named like this. I've been avoiding the abbreviated `Obj` as well